From b68e6164c38ec0e67e966033f087f5182f542150 Mon Sep 17 00:00:00 2001
From: Narendhar Reddy <quic_nare@quicinc.com>
Date: Mon, 4 Sep 2023 00:18:48 +0530
Subject: [PATCH] ath12k: make qdss trace as configurable parameter.

Made qdss trace as a configrurable parameter and is
currently enabled for wkk and disabled for other
targets as there is a known issue.

Signed-off-by: Narendhar Reddy <quic_nare@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.c | 5 ++++-
 drivers/net/wireless/ath/ath12k/hw.c   | 6 ++++++
 drivers/net/wireless/ath/ath12k/hw.h   | 1 +
 3 files changed, 11 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/ath/ath12k/core.c
+++ b/drivers/net/wireless/ath/ath12k/core.c
@@ -1491,7 +1491,10 @@
 		ath12k_hif_ppeds_irq_enable(ab, PPEDS_IRQ_REO2PPE);
 		ath12k_hif_ppeds_irq_enable(ab, PPEDS_IRQ_PPE_WBM2SW_REL);
 #endif
-		ath12k_config_qdss(ab);
+		if (ab->hw_params->en_qdsslog) {
+			ath12k_info(ab, "QDSS trace enabled\n");
+			ath12k_config_qdss(ab);
+		}
 
 		if (ath12k_en_fwlog == true) {
 			ret = ath12k_enable_fwlog(ab);
--- a/drivers/net/wireless/ath/ath12k/hw.c
+++ b/drivers/net/wireless/ath/ath12k/hw.c
@@ -1649,6 +1649,7 @@
 		.en_fwlog = true,
 		.support_umac_reset = false,
 		.is_plink_preferrable = true,
+		.en_qdsslog = true,
 	},
 	{
 		.name = "wcn7850 hw2.0",
@@ -1720,6 +1721,7 @@
 		.en_fwlog = true,
 		.support_umac_reset = false,
 		.is_plink_preferrable = true,
+		.en_qdsslog = true,
 	},
 	{
 		.name = "qcn9274 hw2.0",
@@ -1799,6 +1801,7 @@
 		.en_fwlog = true,
 		.support_umac_reset = true,
 		.is_plink_preferrable = true,
+		.en_qdsslog = true,
 	},
 	{
 		.name = "ipq5332 hw1.0",
@@ -1882,6 +1885,7 @@
 		.support_umac_reset = false,
 		.support_ce_manual_poll=true,
 		.is_plink_preferrable = false,
+		.en_qdsslog = false,
 	},
 	{
 		.name = "qcn6432 hw1.0",
@@ -1958,6 +1962,7 @@
 		.send_platform_model = true,
 		.en_fwlog = false,
 		.support_umac_reset = false,
+		.en_qdsslog = false,
 	},
 };
 
--- a/drivers/net/wireless/ath/ath12k/hw.h
+++ b/drivers/net/wireless/ath/ath12k/hw.h
@@ -231,6 +231,7 @@
 	bool m3_fw_support;
 	bool send_platform_model;
 	bool en_fwlog;
+	bool en_qdsslog;
 	const struct ce_remap *ce_remap;
 	const struct cmem_remap *cmem_remap;
 	bool compact_rx_tlv;
